packages/forms/src/components/ErrorSummary.tsx

```diff
@@ -1,18 +1,141 @@
 import React from "react";
 import { useIntl } from "react-intl";
-import { useFormState } from "react-hook-form";
+import { FieldErrors, FieldValues, useFormState } from "react-hook-form";
+import { ErrorMessage } from "@hookform/error-message";
 
 import {
   Alert,
   ScrollToLink,
   ScrollLinkClickFunc,
   Link,
 } from "@gc-digital-talent/ui";
-import { errorMessages, getLocale } from "@gc-digital-talent/i18n";
+import { commonMessages, errorMessages } from "@gc-digital-talent/i18n";
 import { notEmpty } from "@gc-digital-talent/helpers";
 
 import type { FieldLabels } from "./BasicForm";
 
+/**
+ * Flatten Errors
+ *
+ * Recursively takes the the errors produced by `useFormState`
+ * and flattens them to an array of the field names.
+ *
+ * @param {FieldErrors<FieldValues> } errors - from form state
+ * @param {string} parent - The field name of an existing parent
+ * @returns {string[]}
+ */
+const flattenErrors = (
+  errors: FieldErrors<FieldValues>,
+  parent?: string,
+): string[] => {
+  let errorNames: string[] = [];
+  const parentKey = parent ? `${parent}.` : "";
+
+  Object.keys(errors).forEach((fieldName) => {
+    const fieldError = errors[fieldName];
+    if (fieldError) {
+      // This is a root of a field array, so add it to the key so we are aware later
+      if ("root" in fieldError) {
+        errorNames = [...errorNames, `${fieldName}.root`];
+      }
+      // If it is a field array, loop through, hoisting up field names
+      if (Array.isArray(fieldError)) {
+        fieldError.forEach((subFieldError, index) => {
+          errorNames = [
+            ...errorNames,
+            ...flattenErrors(
+              subFieldError,
+              `${parentKey}${fieldName}.${index}`,
+            ),
+          ];
+        });
+      }
+      // We have an error message so add it to the array (we don't want errors with no message)
+      if ("message" in fieldError) {
+        errorNames = [...errorNames, `${parentKey}${fieldName}`];
+      }
+    }
+  });
```

Comment:
```suggestion
   if (errors)
    Object.keys(errors).forEach((fieldName) => {
      const fieldError = errors[fieldName];
      if (fieldError) {
        // This is a root of a field array, so add it to the key so we are aware later
        if ("root" in fieldError) {
          errorNames = [...errorNames, `${fieldName}.root`];
        }
        // If it is a field array, loop through, hoisting up field names
        if (Array.isArray(fieldError)) {
          fieldError.forEach((subFieldError, index) => {
            errorNames = [
              ...errorNames,
              ...flattenErrors(
                subFieldError,
                `${parentKey}${fieldName}.${index}`,
              ),
            ];
          });
        }
        // We have an error message so add it to the array (we don't want errors with no message)
        if ("message" in fieldError) {
          errorNames = [...errorNames, `${parentKey}${fieldName}`];
        }
      }
    });
```

Reply:
This stopped the error on my end. However, adding a new item doesn't trigger the errors until submit it clicked again which I'm not sure if that's correct behaviour or not.

Reply:
That sounds like it would be ideal actually!